extensions/ecc/circuit/src/weierstrass_chip/double.rs

```diff
@@ -239,29 +239,75 @@ impl<F: PrimeField32, const BLOCKS: usize, const BLOCK_SIZE: usize> InsExecutorE
 
         let is_setup = self.pre_compute_impl(pc, inst, pre_compute)?;
 
-        if is_setup {
-            Ok(execute_e1_setup_impl::<_, _, BLOCKS, BLOCK_SIZE>)
-        } else if let Some(curve_type) = {
+        if let Some(curve_type) = {
             let modulus = &pre_compute.expr.builder.prime;
             let a_coeff = &pre_compute.expr.setup_values[0];
             get_curve_type(modulus, a_coeff)
         } {
-            match curve_type {
-                CurveType::K256 => {
-                    Ok(execute_e1_impl::<_, _, BLOCKS, BLOCK_SIZE, { CurveType::K256 as u8 }>)
+            match (is_setup, curve_type) {
+                (true, CurveType::K256) => {
+                    Ok(
+                        execute_e12_impl::<_, _, BLOCKS, BLOCK_SIZE, { CurveType::K256 as u8 }, true>,
+                    )
                 }
-                CurveType::P256 => {
-                    Ok(execute_e1_impl::<_, _, BLOCKS, BLOCK_SIZE, { CurveType::P256 as u8 }>)
-                }
-                CurveType::BN254 => {
-                    Ok(execute_e1_impl::<_, _, BLOCKS, BLOCK_SIZE, { CurveType::BN254 as u8 }>)
-                }
-                CurveType::BLS12_381 => {
-                    Ok(execute_e1_impl::<_, _, BLOCKS, BLOCK_SIZE, { CurveType::BLS12_381 as u8 }>)
+                (true, CurveType::P256) => {
+                    Ok(
+                        execute_e12_impl::<_, _, BLOCKS, BLOCK_SIZE, { CurveType::P256 as u8 }, true>,
+                    )
                 }
+                (true, CurveType::BN254) => Ok(execute_e12_impl::<
```

Comment:
there was no way to avoid this...?

Reply:
ok I think technically if you made another function that did the routing, you could decrease code, but then you need another function